src/AppKit/NSCollectionViewDelegate.cs

```diff
@@ -0,0 +1,38 @@
+﻿#if !XAMCORE_4_0
+using System;
+using XamCore.Foundation;
+using XamCore.ObjCRuntime;
+
+namespace XamCore.AppKit
+{
+	public partial class NSCollectionViewDelegate
+	{
+		[Obsolete ("Use 'ValidateDropOperation (NSCollectionView collectionView, NSDraggingInfo draggingInfo, ref NSIndexPath proposedDropIndexPath, ref NSCollectionViewDropOperation proposedDropOperation)' instead.")]
+		[Mac (10, 11)]
+		public virtual NSDragOperation ValidateDrop (NSCollectionView collectionView, NSDraggingInfo draggingInfo, out NSIndexPath proposedDropIndexPath, out NSCollectionViewDropOperation proposedDropOperation)
+		{
+			throw new InvalidOperationException ("Use 'ValidateDropOperation (NSCollectionView collectionView, NSDraggingInfo draggingInfo, ref NSIndexPath proposedDropIndexPath, ref NSCollectionViewDropOperation proposedDropOperation)' instead.");
```

Comment:
Could these methods work in any scenario with the `out` parameter? If so, we should continue making it work by calling the `ref` overload instead of throwing an exception.

Same goes for the other methods below.

Reply:
From what I can tell they do not.  The ValidateDrop method is called when you drag an object in the collection view.  When you drag over a location, it calls Validatedrop to determine if it can be dropped there.  With the 'out' parameter, the method is never called in the first place from what I've seen.

Reply:
I suppose there's no real reason I couldn't make it call the ref version anyway